dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/DefaultProgramRuleEntityMapperService.java

```diff
@@ -551,4 +554,32 @@ private DataItem getDisplayName( ProgramRuleVariable prv )
             .valueType( getItemValueType( dataElement.getValueType() ) )
             .build();
     }
+
```

Comment:
This method is quite difficult to follow.

Could you add a guard clause like
```
if ( prv.getUseCodeForOptionSet() )
{
    return List.of();
}
```

- Do not use a builder for Option object
- Do not use Optional.ofNullable to check existence of something, put `prv.getDataElement().getOptionSet()` in the if like `if ( prv.hasDataElement() && prv.getDataElement().getOptionSet() )`
- Same for attributes

Reply:
I can try that. but probably sonarcloud would complain

Reply:
resolved